### PR TITLE
Migrate Function Controller to use Kubebuilder 2

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- Migrate function-controller to use Kubebuilder 2:
  - update dependencies to match those provided by kubebuilder (controller-runtime)
  - update Makefile to use generic version from `common` folder

Changes in `resources` folder will be done in separate PR

**Related issue(s)**

See also #6738

### CODE REVIEW

File: components/function-controller/config/default/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -22,3 +22,4 @@ spec:
       - name: manager
         args:
         - "--metrics-addr=127.0.0.1:8080"
+        - "--enable-leader-election"
```
Comment (aerfio) [inline components/function-controller/config/default/manager_auth_proxy_patch.yaml]: Do we need this?

File: components/function-controller/config/manager/manager.yaml [hunk 2]
```diff
@@ -1,87 +1,41 @@
...
+      annotations:
+        sidecar.istio.io/inject: "false"
     spec:
       containers:
-      - image: github.com/kyma-project/kyma/components/function-controller/cmd/manager
-        imagePullPolicy: Always
+      - command:
+        - /manager
+        args:
+        - --enable-leader-election
```
Comment (aerfio) [inline components/function-controller/config/manager/manager.yaml]: do we need this?

File: components/function-controller/config/manager/manager.yaml [hunk 3]
```diff
@@ -1,87 +1,41 @@
...
-      controller-tools.k8s.io: "1.0"
-  serviceName: controller-manager-service
+  replicas: 1
   template:
     metadata:
       labels:
         control-plane: controller-manager
-        controller-tools.k8s.io: "1.0"
+      annotations:
+        sidecar.istio.io/inject: "false"
```
Comment (aerfio) [inline components/function-controller/config/manager/manager.yaml]: .

File: components/function-controller/hack/test-cluster.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+# easy script to setup
+# TODO add basic trap on err
+
```
Comment (m00g3n) [inline components/function-controller/hack/test-cluster.sh]: this script hast to be called from function-controller directory can wee have info about it;
can we have a short info on potential collision with kind in different version, that could be visible via PATH?

File: components/function-controller/hack/test-cluster.sh [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    kind create cluster --name "${1}" --image "${image}" --config "${SCRIPT_DIR}/cluster-config.yaml" --wait 3m
+
+    echo "Cluster created"
+}
+
+istio::download_istioctl(){
+    local -r destination_dir="${1}"
+    echo "Downloading istio"
+    curl -L https://istio.io/downloadIstio | sh - \
+    && chmod +x "istio-1.4.3/bin/istioctl" \
```
Comment (m00g3n) [inline components/function-controller/hack/test-cluster.sh]: we can parametrize version

File: components/function-controller/hack/test-cluster.sh [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+tekton::install(){
+    kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin
+    kubectl apply -f "https://storage.googleapis.com/tekton-releases/pipeline/previous/${TEKTON_VERION}/release.yaml"
+}
+
+cert-manager::install(){
+    kubectl create namespace cert-manager
+    kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.12.0/cert-manager.yaml
```
Comment (m00g3n) [inline components/function-controller/hack/test-cluster.sh]: we can parametrize version

File: components/function-controller/.gitignore [hunk 7]
```diff
@@ -23,3 +23,5 @@ vendor/
 *.swp
 *.swo
 *~
+
+function-controller
```
Comment (m00g3n) [inline components/function-controller/.gitignore]: we may think of a dedicated directory for a build files

File: components/function-controller/config/default/manager_webhook_patch.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
```
[the remark this answers is not in the document]
Comment (aerfio) [reply components/function-controller/config/default/manager_webhook_patch.yaml]: already handled 👍

File: components/function-controller/.gitignore (hunk 7 above)
Comment (aerfio) [reply components/function-controller/.gitignore]: generic makefile defines this directory

**Pull request merged**